### ISSUE TITLE
In-memory data source for non-point geometries

### ISSUE THREAD

**Issue by guigrpa** [mapnik/node-mapnik-sample-code#8]
Is there an example of an in-memory data source with non-point geometries? I couldn't find any. Is this even possible? I noticed from the code that there are plans to support WKT format, but it's not yet implemented.

On the other side, since standard Datasource objects always read data from file, is there a work-around for injecting non-point geometries into Mapnik programmatically?

**Comment by springmeyer** [comment]
Yes, see https://github.com/mapnik/node-mapnik/issues/779 and also https://github.com/mapnik/node-mapnik/issues/779

**Comment by guigrpa** [comment]
Is there an example for using this inline geojson string?

**Comment by springmeyer** [comment]
Yes, sorry one of the links above was meant to be to https://github.com/mapnik/python-mapnik/issues/126#issuecomment-326819222. There you can see a link other code which offers an example.

**Comment by guigrpa** [comment]
Thanks a lot! I'll have a look

**Comment by guigrpa** [comment]
Inlining GeoJSON works perfectly, and in my experiment runs >10x faster than via the OGR plugin. Thanks a lot for the insight!

Any similar solution for rasters? Is it possible in general to create a `Mapnik.Datasource` with an arbitrary `type` and suitable `inline` fields?
